Develop

DanaBass opened the pull request.

DanaBass committed:
Добавила файл conftest для фикстур.

DanaBass committed:
Создала класс для раздела "Вопросы о важном".
В файле с тестами его протестировала.

DanaBass committed:
Описала базовые методы для проекта.

DanaBass committed:
Написала фикстуру для драйвера.

DanaBass committed:
Создала класс для главной страницы и успешно его протестировала с помощью параметризации.

DanaBass committed:
Изменила все Directorys на Python Packages.

DanaBass committed:
Добавила тесты и вспомогательный код.

DanaBass committed:
Добавила декораторы Allure и добавила описание README.
Добавила обработку ошибки связанную с кликом по кнопке.

DanaBass committed:
Добавила сгенерированный allure-отчет и файл requirements.txt

klushcheva commented on conftest.py:
```diff
@@ -0,0 +1,15 @@
+@pytest.fixture()
+def driver():
+    driver = webdriver.Firefox()
+    driver.get('https://qa-scooter.praktikum-services.ru/')
```
Можно лучше: выносить urlы в переменные

klushcheva commented on conftest.py:
```diff
@@ -0,0 +1,15 @@
+def driver():
+    driver = webdriver.Firefox()
+    driver.get('https://qa-scooter.praktikum-services.ru/')
+    confirm_button = WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, "//button[@id='rcc-confirm-button' and text()='да все привыкли']")))
```
Нужно поправить: все локаторы лежат в файле с локаторами

klushcheva commented on data/data.py:
```diff
@@ -0,0 +1,31 @@
+        7: "Да, обязательно. Всем самокатов! И Москве, и Московской области."
+    }
+
+class OrderFormData:
```
Это лишний класс

klushcheva commented on pages/base_page.py:
```diff
@@ -0,0 +1,60 @@
+    @allure.step("Находим элемент по локатору: {locator}")
+    @allure.title("Поиск элемента по локатору")
+    def find_element_with_wait(self, locator):
+        return WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(locator))
```
Можно лучше:Можно лучше: во всех методах, где используется WebDriverWait, ожидание можно вынести как параметр метода - поставить дефолтное значение и менять при необходимости. Либо вынести это как атрибут в init: self.time = 10

klushcheva commented on tests/test_main_page.py:
```diff
@@ -0,0 +1,20 @@
+
+class TestMainPage:
+
+    @pytest.mark.parametrize("num", [
```
Отлично

DanaBass committed:
Исправила замечания по ревью:
1. Создала файл urls.py для хранения URLS.
2. Перенесла локатор в отдельный файл - теперь все локаторы лежат в одном месте.
3. Удалила лишний класс OrderFormData. Теперь тестовые данные для формы заказа хранятся в словаре класса OrderForm.
4. В методах класса BasePage, где необходим таймаут, ввела дополнительный параметр timeout со значением по умолчанию. Теперь вызывающий код может отдавать своё необходимое время для ожидания.
